src/main/java/duke/frontend/MainWindow.java

```diff
@@ -0,0 +1,111 @@
+package duke.frontend;
+
+import duke.Duke;
+import duke.parser.IncorrectArgumentsException;
+import duke.parser.IncorrectFileFormatException;
+import javafx.fxml.FXML;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+import java.io.FileNotFoundException;
+import java.nio.file.InvalidPathException;
+
+/**
+ * Controller for MainWindow. Provides the layout for the other controls.
+ */
+public class MainWindow extends AnchorPane {
+	@FXML
+	private ScrollPane scrollPane;
+	@FXML
+	private VBox dialogContainer;
+	@FXML
+	private TextField userInput;
+	private Duke duke;
+	private Image userImage;
+	private Image dukeImage;
+	private Image jigglypuff;
+	
+	/**
+	 * Initializes the chat bot GUI components, scroll pane and images of bot.
+	 */
+	@FXML
+	public void initialize() {
+		scrollPane.vvalueProperty().bind(dialogContainer.heightProperty());
+		userImage = new Image(this.getClass().getResourceAsStream("/images/chatbot.png"));
+		dukeImage = new Image(this.getClass().getResourceAsStream("/images/robot.png"));
+		jigglypuff = new Image(this.getClass().getResourceAsStream("/images/avatar.png"));
+	}
+	
+	/**
+	 * Set up Duke for main GUI, prepare the task list for first load.
+	 */
+	public void setDuke(Duke d) {
+		duke = d;
+		String loadingErrorMessage = "";
+		boolean isErrorOccurred = false;
+		
+		// Perform loading of tasks from disk
```

Comment:
is there a way you could avoid loading tasks from disk within setDuke()? it seems like this chunk of code may be repeated from Duke.java.